pkg/process_manager.go

```diff
@@ -19,45 +25,172 @@ const (
 
 type ProcessStatus struct {
 	// ExitStatus is only valid if `State` is `EXITED`.
-	ExitStatus int32
+	ExitStatus int
 	State      ProcessState
 }
 
 type processStream struct {
 	// TODO
 }
 
-func (*processStream) Read(p []byte) (n int, err error) {
+func (s *processStream) Read(p []byte) (n int, err error) {
 	panic("TODO")
 }
 
+type process struct {
+	// Not 100% sure if this is necessary since the docs don't mention anything
+	// about thread safety for `exec.Cmd` functions as far as I can tell, so
+	// better safe than sorry.
+	cmdLock *sync.RWMutex
+	cmd     *exec.Cmd
+}
+
 type ProcessManager struct {
-	// TODO
+	processesLock sync.RWMutex
+	processes     []process
 }
 
-func (*ProcessManager) Start() uint32 {
-	panic("TODO")
+var ErrUnknownId = errors.New("unknown process ID")
```

Comment:
```suggestion
var ErrUnknownID = errors.New("unknown process ID")
```

https://go.dev/wiki/CodeReviewComments#initialisms

Reply:
Done.